app/controllers/api/v2/hosts_controller.rb

```diff
@@ -203,6 +203,21 @@ def facts
         render_message(e.to_s, :status => :unprocessable_entity)
       end
 
+      api :POST, "/hosts/:id/rebuild_config", N_("Rebuild orchestration config (TFTP, DNS and DHCP)")
+      param :id, :identifier_dottable, :required => true
+      def rebuild_config
+        result = @host.recreate_config
+        failures = []
+        result.each_pair do |key, value|
```

Comment:
A little more Ruby-ish might be to do something like: `result.reject { |k,v| v.blank? }.keys.map(&:to_s)`